dp/generate_climos.py

```diff
@@ -39,17 +39,96 @@
 cdo = Cdo()
 
 
+def input_check(filepath, climo):
+    '''
+    This function runs general checks on the given input arguments filepath and climo.
+    First, it checks if the given input file path is a vaild NetCDF file. It returns
+    NetCDF dataset of the input file. Ohterwise, it halts the program. Next, it checks
+    if the input NetCDF dataset has a set of standard climo periods by using climo_periods
+    property. It returns the periods if found, outherwise it halts the program.
+    '''
+    logger.info("")
+    logger.info("Processing: {}".format(filepath))
+
+    try:
+        input_file = CFDataset(filepath)
+    except OSError as e:
+        logger.info("{}: {}".format(e.__class__.__name__, e))
+        return
```

Comment:
You are returning nothing (no value) here, but the normal return returns a pair. This will cause the caller some pretty serious problems, since it is going to be expecting a pair. Suggest `return None, []` -- no file, empty period list.

In general, we avoid returning different types of results from the same function. And especially we avoid returning different types when the caller can't tell in advance what to expect, as here.